proposed/phpdoc.md

```diff
@@ -0,0 +1,2214 @@
+#### Description
+
+The @package tag can be used as a counterpart or supplement to Namespaces.
+Namespaces provide a functional subdivision of "Structural Elements" where the
+@package tag can provide a *logical* subdivision in which way the elements can
+be grouped with a different hierarchy.
+
+If, across the board, both logical and functional subdivisions are equal it
+is NOT RECOMMENDED to use the @package tag, to prevent maintenance overhead.
+
+Each level in the logical hierarchy MUST separated with a backslash (`\`) to
+be familiar to Namespaces. A hierarchy MAY be of endless depth but it is
+RECOMMENDED to keep the depth at less or equal than six levels.
+
+Please note that the @package applies to different "Structural Elements"
+depending where it is defined.
+
+1. If the @package is defined in the *file-level* DocBlock then it only applies
+   to the following elements in the applicable file:
+    * global functions
+    * global constants
+    * global variables
+    * requires and includes
+
+2. If the @package is defined in a *namespace-level* or *class-level* DocBlock
+   then the package applies to that namespace, class or interface and their
+   contained elements.
+   This means that a function which is contained in a namespace with the
+   @package tag assumes that package.
+
+This tag MUST NOT occur more than once in a "DocBlock".
+
+#### Examples
+
+```php
+/**
+ * @package PSR\Documentation\API
+ */
+```
+
+### 8.13. @param
+
+The @param tag is used to document a single parameter of a function or method.
+
+#### Syntax
+
+    @param ["Type"] [name] [<description>]
+
+#### Description
+
+With the @param tag it is possible to document the type and function of a
+single parameter of a function or method. When provided it MUST contain a
+"Type" to indicate what is expected; the description on the other hand is
+OPTIONAL yet RECOMMENDED. For complex structures such as option arrays it is
+RECOMMENDED to use an "Inline PHPDoc" to describe the option array.
+
+The @param tag MAY have a multi-line description and does not need explicit
+delimiting.
+
+It is RECOMMENDED when documenting to use this tag with every function and
+method. Exceptions to this recommendation are:
+
+This tag MUST NOT occur more than once per parameter in a "PHPDoc" and is
+limited to "Structural Elements" of type method or function.
+
+#### Examples
+
+```php
+/**
+ * Counts the number of items in the provided array.
+ *
+ * @param mixed[] $array Array structure to count the elements of.
+ *
+ * @return int Returns the number of elements.
+ */
+function count(array $items)
+{
+    <...>
+}
+```
+
+The following example demonstrates the use of an "Inline PHPDoc" to document
+an option array with 2 elements: 'required' and 'label'.
+
+```php
+/**
+ * Initializes this class with the given options.
+ *
+ * @param array $options {
+ *     @var bool   $required Whether this element is required
+ *     @var string $label    The display name for this element
+ * }
```

Comment:
In that case I don't understand why an assocative array should have a representation in PHPDoc and not a tuple (i.e. a non-associative array), e.g.

````
/**
 * @param array $something {
 *   @var bool This is the first element in the tuple
 *   @var integer This is the second
 * }
 */
````